extension/source/background/TransactionsController.ts

```diff
@@ -115,15 +110,15 @@ export default class TransactionsController {
 
     requestTransaction: async ({ params }) => {
       const id = await this.InternalRpc().createTransaction(...params);
-      const result = await this.InternalRpc().promptUser(id);
+      const userAction = await this.InternalRpc().promptUser(id);
+
+      await this.InternalRpc().updateTransactionStatus(id, userAction);
 
-      if (
-        result === TransactionStatus.APPROVED ||
-        result === TransactionStatus.REJECTED
-      ) {
```

Comment:
We enforce that `promptUser` always responds with one of these, so no need to check it again.